File: leetcode/cpp/551_Student_Attendance_Record_I.cpp
#include <string>
using namespace std;
class Solution {
public:
	bool checkRecord(string s) {
		int L_cnt = 0, A_cnt = 0;
		for(char c : s) {
			if (c == 'A') {
				++A_cnt;
				L_cnt = 0;
				if (A_cnt == 2) return false;
			} else if (c == 'L') {
				++L_cnt;
				if (L_cnt == 3) return false;
			} else
				L_cnt = 0;
		}

		return true;
	}
};

File: leetcode/cpp/551_Student_Attendance_Record_I.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        lcnt, acnt = 0, 0
        for c in s:
            if c == 'A':
                acnt += 1
                if acnt == 2: return False
                lcnt = 0
            elif c == 'L':
                lcnt += 1
                if lcnt == 3: return False
            else:
                lcnt = 0

        return True
